src/iexamine.cpp

```diff
@@ -974,23 +974,35 @@ void iexamine::slot_machine(player *p, map*, const tripoint&)
 void iexamine::safe(player *p, map *m, const tripoint &examp)
 {
     if (!p->has_amount("stethoscope", 1)) {
-        add_msg(m_info, _("You need a stethoscope for safecracking."));
-        return;
+        if (query_yn(_("The safe is locked. Input a random combination?"))) {
+            p->moves -= 100;
+            // one_in(30^3) chance of guessing
+            if (one_in(27000)) {
```

Comment:
With a chance this low, it's more of an anti-feature than a feature - it adds another query to menu and is only actually usable with a macro system (which doesn't come with the game).
Either move it into a long action that repeats the roll until success/cancel or remove it completely. You could integrate it into the cracking action.